paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/CdcRecordStoreWriteOperator.java

```diff
@@ -86,10 +104,18 @@ public void processElement(StreamRecord<CdcRecord> element) throws Exception {
             write.replace(table);
         }
 
-        try {
-            write.write(optionalConverted.get());
-        } catch (Exception e) {
-            throw new IOException(e);
+        if (!optionalConverted.isPresent()) {
+            if (skipCorruptRecord) {
+                LOG.warn("Skipping corrupt or unparsable record {}", record);
```

Comment:
just a note that this might leak sensitive data to the log system. maybe we can log some metadata about the record instead of the full record?